### PR TITLE
Add mirror flag for unpublished stages

### PR DESCRIPTION
**Pull request by jesusbv**
## Description

  do not raise an exception if there is missing
  info due to being in alpha or beta stage

This references bsc#1180018

## Change Type

*Please select the correct option.*

- [X] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [X] I have verified that my code follows RMT's coding standards with `rubocop`.
- [X] I have reviewed my own code and believe that it's ready for an external review.
- [X] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [X] RMT's test coverage remains at 100%.
- [X] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/178825/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged f0f11174f1287e9ce7faa68c6d994213095a1329 into f33b1166afafc06baaf402e40da01c3f9c56e435)

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><p>The version of RMT is not consistent. These files must specify the same version:</p>
<ul>
  <li><code>lib/rmt.rb</code> specifies <code>2.6.7</code></li>
  <li><code>package/obs/rmt-server.spec</code> specifies <code>2.6.6</code></li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/180522/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 54e959599e88289db4afe83e4227fc0af9c5754a into f33b1166afafc06baaf402e40da01c3f9c56e435)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/180523/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6bcff80813c2bdd9203e80e1b9001af904a1f896 into f33b1166afafc06baaf402e40da01c3f9c56e435)

**Review by gionaufal — APPROVED** [review]
lgtm!

**Comment by jesusbv** [comment]
Hi @tmuntaner and @lcaparroz,

if no objection, I will merge this then

File: lib/rmt/cli/mirror.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class RMT::CLI::Mirror < RMT::CLI::Base
+  class_option :do_not_raise_unpublished, desc: _('Do not fail the command if info is in alpha or beta stage'), type: :boolean, required: false
```
Comment (lagartoflojo) [inline lib/rmt/cli/mirror.rb:2]: > Do not fail the command if **info** is in...

What do you mean by "info"?
I would rather consider that a _product_ is in alpha or beta stage.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/184061/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests.
Failed fast in project rmt on step docker-compose-run-tests.
Failed fast in project rmt on step docker-compose-run-tests.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 79760a89cb8d08733391ab418d3237865a8fbc61 into 023f61747d6a2b527cbc0ac4f0a83d9845a423a3)

File: lib/rmt/cli/mirror.rb (hunk 1 above)
Comment (jesusbv) [reply lib/rmt/cli/mirror.rb]: Done

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/184088/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 9337d5800373719c9a23e6620842fbdf33cd0b80 into 023f61747d6a2b527cbc0ac4f0a83d9845a423a3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/184203/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 9337d5800373719c9a23e6620842fbdf33cd0b80 into b647997301d9366250e0e204efb300c8cd74e7d4)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/184204/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged b5988ac358fa63391949e2655e8ca43b14d46c8e into b647997301d9366250e0e204efb300c8cd74e7d4)

**Pull request merged**